src/psyclone/tests/test_files/dynamo0p3/columnwise_op_asm_field_kernel_mod.F90

```diff
@@ -100,22 +130,24 @@ subroutine columnwise_op_asm_field_kernel_code(cell,              &
                                             gamma_p,           &
                                             undf, ndf,         & ! any_space_1
                                             dofmap_field,      &
-                                            ndf_lma_from,      & ! any_space_2
                                             column_banded_dofmap_to, &
+                                            ndf_lma_from,      & ! any_space_2
                                             column_banded_dofmap_from)
 
     implicit none
     
     ! Arguments
-    integer(kind=i_def), intent(in) :: cell,  nlayers, ncell_3d, ncell_2d
-    integer(kind=i_def), intent(in) :: ndf_lma_to, ndf_lma_from
-    integer(kind=i_def), intent(in) :: ndf_to, ndf_from
-    real(kind=r_def), dimension(ndf_lma_to,ndf_lma_from,ncell_3d), intent(in) :: local_stencil
+    integer(kind=i_def), intent(in) :: cell,  nlayers, ncell_2d, ncell_3d
+    integer(kind=i_def), intent(in) :: ndf_lma_from
+    integer(kind=i_def), intent(in) :: undf, ndf
+    real(kind=r_def), dimension(undf), intent(in) :: field
+    real(kind=r_def), dimension(:,:,:), intent(in) :: local_stencil
     integer(kind=i_def), intent(in) :: nrow, ncol, bandwidth
-    real(kind=r_def), dimension(bandwidth,nrow,ncell_2d), intent(out) :: columnwise_matrix
+    real(kind=r_def), dimension(:,:,:), intent(out) :: columnwise_matrix
```

Comment:
Did the compiler not like having explicit dimensions for the array? Ditto `local_stencil`?

Reply:
I'm not sure. I cut it down from a Met Office example so either it was like that or I just didn't bother adding them. I've replaced the argument list with what we produce from kernelgen as the arguments they provided differ from the ones we generate!